packages/react-native-sdk/src/hooks/useScreenShareButton.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  getLogger,
+  hasScreenShare,
+  OwnCapability,
+} from '@stream-io/video-client';
+import { useCall, useCallStateHooks } from '@stream-io/video-react-bindings';
+import React, { useRef, useEffect } from 'react';
+import { Platform, findNodeHandle, NativeModules } from 'react-native';
+import { usePrevious } from '../utils/hooks';
+import { useIsIosScreenshareBroadcastStarted } from './useIsIosScreenshareBroadcastStarted';
+
+// ios >= 14.0 or android - platform restrictions
+const CanDeviceScreenShare =
+  (Platform.OS === 'ios' &&
+    // @ts-ignore
+    Number.parseInt(Platform.Version.split('.')[0], 10) >= 14) ||
+  Platform.OS === 'android';
+
+export const useScreenShareButton = (
+  /**
+   * Ref of the ScreenCapturePickerView component.
+   *
+   */
+  screenCapturePickerViewiOSRef: React.MutableRefObject<any>,
+  /**
+   * Handler to be called when the screen-share has been started.
+   *
+   */
+  onScreenShareStartedHandler?: () => void,
+  /**
+   * Handler to be called when the screen-share has been stopped.
+   *
+   */
+  onScreenShareStoppedHandler?: () => void,
+  /**
+   * Handler to be called when the permissions to stream screen share media is missing
+   *
+   */
+  onMissingScreenShareStreamPermission?: () => void
+) => {
+  const call = useCall();
+  const { useLocalParticipant, useCallSettings, useOwnCapabilities } =
+    useCallStateHooks();
+  const callSettings = useCallSettings();
+  const ownCapabilities = useOwnCapabilities();
+  const hasScreenSharingPermissions = ownCapabilities?.includes(
+    OwnCapability.SCREENSHARE
+  );
+  const isScreenSharingEnabledInCall = callSettings?.screensharing.enabled;
+
+  const onScreenShareStartedHandlerRef = useRef(onScreenShareStartedHandler);
+  onScreenShareStartedHandlerRef.current = onScreenShareStartedHandler;
+  const onScreenShareStoppedHandlerRef = useRef(onScreenShareStoppedHandler);
+  onScreenShareStoppedHandlerRef.current = onScreenShareStoppedHandler;
+
+  const iosScreenShareStartedFromSystem = useIsIosScreenshareBroadcastStarted();
+  const prevIosScreenShareStartedFromSystem = usePrevious(
+    iosScreenShareStartedFromSystem
+  );
+
+  const localParticipant = useLocalParticipant();
+  const hasPublishedScreenShare =
+    localParticipant && hasScreenShare(localParticipant);
+
+  // listens to iOS screen share broadcast started event from the system
+  useEffect(() => {
+    const run = async () => {
```

Comment:
As this function is marked as `async` perhaps you want to `await` some async functions inside? Perhaps `call.screenShare.enable()` and `call.screenShare.disable()`?

Reply:
btw, awesome stuff - switching to `call.screenShare` APIs 🚀